compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -202,10 +207,16 @@ private Path emitBala(Path filePath) {
 
         List<PlatformLibrary> platformLibraries = new ArrayList<>();
         for (Map<String, Object> dependency : javaPlatform.dependencies()) {
-            String dependencyFilePath = (String) dependency.get(JarLibrary.KEY_PATH);
             String artifactId = (String) dependency.get(JarLibrary.KEY_ARTIFACT_ID);
             String version = (String) dependency.get(JarLibrary.KEY_VERSION);
             String groupId = (String) dependency.get(JarLibrary.KEY_GROUP_ID);
+
+            String dependencyFilePath = (String) dependency.get(JarLibrary.KEY_PATH);
```

Comment:
Do we have tests to test this path ?

Reply:
There's an integration test called `MavenTestCase` covering this scenario and unit tests in `MavenResolverTest` class. These unit tests are testing `Utils.getJarPath` functions which have used inside `getPlatformLibPath` method in `JBallerinaBackend`.